File: smart pay app/app/src/main/java/com/smartpay/application/MainActivity.java
package com.smartpay.application;

import static com.smartpay.application.MotoTransactionActivity.MOTO_TYPE;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setToolbar(getString(R.string.welcome), false);

    }

    public void startSingleMotoTransaction(View view) {
        Intent intent = new Intent(this, MotoTransactionActivity.class);
        if (view.getId() == R.id.tv_single_moto)
            intent.putExtra(MOTO_TYPE, getString(R.string.single_moto));
        else intent.putExtra(MOTO_TYPE, getString(R.string.recurring_moto));
        startActivity(intent);
    }
}

File: smart pay app/app/src/main/java/com/smartpay/application/MainActivity.kt
package com.smartpay.application

import android.os.Bundle
import android.content.Intent
import android.content.IntentFilter
import android.view.View
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.smartpay.application.common.Constants
import com.smartpay.application.common.Constants.Companion.TLV_REQUEST_PROCESS
import com.smartpay.application.fake.payment.switch.FakeRequestReceiver
import com.smartpay.application.moto.transaction.MotoTransactionActivity
import java.security.AccessController.getContext

class MainActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setToolbar(getString(R.string.welcome), false)
        LocalBroadcastManager.getInstance(this)
            .registerReceiver(FakeRequestReceiver(), IntentFilter(TLV_REQUEST_PROCESS));
    }

    fun startMotoTransaction(view: View) {
        val intent = Intent(this, MotoTransactionActivity::class.java)
        if (view.id == R.id.tv_single_moto) intent.putExtra(
            Constants.IS_SINGLE_MOTO_TYPE,
            true
        ) else intent.putExtra(
            Constants.IS_SINGLE_MOTO_TYPE,
            false
        )
        startActivity(intent)
    }
}